src/compiler/resolutionCache.ts

```diff
@@ -14,7 +14,7 @@ namespace ts {
         removeResolutionsOfFile(filePath: Path): void;
         removeResolutionsFromProjectReferenceRedirects(filePath: Path): void;
         setFilesWithInvalidatedNonRelativeUnresolvedImports(filesWithUnresolvedImports: ESMap<Path, readonly string[]>): void;
-        createHasInvalidatedResolution(forceAllFilesAsInvalidated?: boolean): HasInvalidatedResolution;
+        createHasInvalidatedResolution(userProvidedHasInvalidatedResolution?: HasInvalidatedResolution): HasInvalidatedResolution;

```

Comment:
I think it would find it clearer if this were just `hasInvalidatedResolution: HasInvalidatedResolution` because it seems like the this can be set even when the user didn't provide an implementation (it's `returnTrue` if `userProvidedResolution`?).  So I'd just pass `returnFalse` if no other value is available.